yarn-project/aztec-nr/aztec/src/log.nr

```diff
@@ -14,9 +14,25 @@ fn emit_encrypted_log<N>(
 }
 
 fn emit_unencrypted_log<T>(
+    context: &mut PublicContext,
+    log: T,
+) {
+    let contract_address = context.this_address();
+    let event_selector = 5; // TODO: compute actual event selector.
+    let _ = oracle::logs::emit_unencrypted_log(contract_address, event_selector, log);
+    // context.accumulate_unencrypted_logs(log);
+}
+
+// TODO: We might want to remove this since emitting unencrypted logs from private functions is violating privacy.
+// --> might be a better approach to force devs to make a public function call that emits the log if needed then
+// it would be less easy to accidentally leak information.
+// If we decide to keep this function around would make sense to wait for traits and then merge it with emit_unencrypted_log.
+fn emit_unencrypted_log_from_private<T>(
```

Comment:
Did you create this function so that it is easy to identify what to remove if we choose to do so?

Reply:
It was one of the reasons. But the main one was that if I wanted to have 1 function for use in both private and public contexts I would have to use the Context wrapper and that seemed like too much of a boilerplate given that the:
Instead of this:
`emit_unencrypted_log(&mut context, "Coins transferred");
`
I would have to use:
`emit_unencrypted_log(Context::public(&mut context), "Coins transferred");
`